pkg/cmd/server/server.go

```diff
@@ -235,6 +238,11 @@ func (c *Config) Complete() (RunnableServer, error) {
 		c.UnaryMiddleware, c.StreamingMiddleware = DefaultMiddleware(log.Logger, c.GRPCAuthFunc, !c.DisableVersionResponse, dispatcher, ds)
 	}
 
+	permSysConfig := v1svc.PermissionsServerConfig{
+		MaxPreconditionsCount: defaultIfZero(c.MaximumPreconditionCount, 1000),
```

Comment:
You shouldn't need the `defaultIfZero` in this case - the cli flag bindings set the default values for you.

Reply:
They do, but if I manually create the config they'll start at 0, so I want to be absolutely sure